contracts/contracts/GrantRoundManager.sol

```diff
@@ -32,35 +24,70 @@ contract GrantRoundManager {
   /// @notice Address of the ERC20 token in which donations are made
   IERC20 public immutable donationToken;
 
+  /// @notice Used during donations to temporarily store swap output amounts
+  mapping(IERC20 => uint256) internal swapOutputs;
```

Comment:
> Mappings can only have a data location of `storage`

([source](https://docs.soliditylang.org/en/v0.8.7/types.html#mapping-types))

Given that, it costs 20k gas to write and you get a refund of 19.9k when clearing in the same transaction as the initial write, so it ends up being cheap ([source](https://github.com/wolflo/evm-opcodes/blob/main/gas.md#a7-sstore)). Open to other ideas on how to do this, though this seemed the cleanest

Reply:
🤦  Duh I should have known that. I think our only option would be to use something like dynamic arrays of tuples. This would have O(N) lookup time. I'm curious where the break-even point would be on N, but given we can get refunded for the storage slots it's probably pretty low. It'd also be a much uglier implementation.

Another option would be to require-by-convention that donations be grouped by token. Then you could do everything by array index, with inner while loops that increment the index when the token changes. Again, more complex implementation. Also  adds work on the frontend (albeit pretty minor). Might be worth exploring, but feels like premature optimization at the moment.

Let's keep it as-is and revisit whenever we get around to gas golfing.